File: app/src/main/java/com/tomaschlapek/tcbasearchitecture/realm/RUser.java
package com.tomaschlapek.tcbasearchitecture.realm;

import io.realm.RealmObject;
import io.realm.annotations.PrimaryKey;

/**
 * Class holds information about certain user.
 */
public class RUser extends RealmObject {

  /* Attributes ***********************************************************************************/

  @PrimaryKey
  private String token;

  private String nickName;
  private int credits;
  private String avatar;


  /* Public methods *******************************************************************************/

  /**
   * Default constructor.
   */
  public RUser() {
  }

  /*public RUser(UserInfoResponse userInfoResponse) {
      this.nickName = userInfoResponse.getUserNickName();
      this.credits = userInfoResponse.getUserCredits();
      this.token = userInfoResponse.getToken();
      this.avatar = userInfoResponse.getAvatar();
  }*/

  /* Getters / Setters ****************************************************************************/

  public String getNickName() {
    return nickName;
  }

  public void setNickName(String nickName) {
    this.nickName = nickName;
  }

  public int getCredits() {
    return credits;
  }

  public void setCredits(int lastName) {
    this.credits = lastName;
  }

  public String getToken() {
    return token;
  }

  public void setToken(String token) {
    this.token = token;
  }

  public String getAvatar() {
    return avatar;
  }

  public void setAvatar(String avatar) {
    this.avatar = avatar;
  }
}

File: app/src/main/kotlin/com/tomaschlapek/tcbasearchitecture/realm/RUser.kt
package com.tomaschlapek.tcbasearchitecture.realm

import com.google.firebase.auth.FirebaseUser
import io.realm.RealmObject
import io.realm.annotations.PrimaryKey

/**
 * Class holds information about certain user.
 */
open class RUser() : RealmObject() {
  constructor(firebaseUser: FirebaseUser) : this() {
    this.id = firebaseUser.uid
    this.email = firebaseUser.email
    this.nickName = firebaseUser.displayName
    this.verified = firebaseUser.isEmailVerified
  }

  /* Attributes ***********************************************************************************/

  @PrimaryKey
  var id: String? = null

  var email: String? = null
  var nickName: String? = null
  var verified: Boolean = false
}